tests/utils/test_modeling_utils.py

```diff
@@ -460,6 +461,24 @@ def test_model_from_config_torch_dtype_str(self):
         with self.assertRaises(ValueError):
             model = AutoModel.from_pretrained(TINY_T5, torch_dtype="int64")
 
+    @slow
+    def test_model_from_pretrained_meta_device(self):
+        def is_on_meta(model_id, dtype):
+            with torch.device("meta"):
+                model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=dtype)
+                return all(value.device.type == "meta" for value in model.state_dict().values())
+
+        result = {}
+        expected = {}
+        model_ids = ("fxmarty/tiny-llama-fast-tokenizer", "fxmarty/small-llama-testing")
+        dtypes = (None, "auto", torch.float16)
+
+        for model_id, dtype in itertools.product(model_ids, dtypes):
+            result[(model_id, dtype)] = is_on_meta(model_id, dtype)
+            expected[(model_id, dtype)] = True
+
+        assert result == expected
```

Comment:
```suggestion
            self.assertTrue(is_on_meta(model_id, dtype))
```

Reply:
This was my first instinct as well, and it is definitely more readable. However, if there are multiple asserts which fail, this would only catch the first failure, thus making it necessary to iterate by running the test several times and fixing one by one. I have fixed thousands of tests, so I prefer the approach of one assert per test, but I can also live with proposed solution - so please feel free to keep the last commit or remove it :)

Reply:
Makes sense! Both sounds good to me!